docs/MAINTAINERS.adoc

```diff
@@ -0,0 +1,162 @@
+* Breakdown the commit history or squashing PRs
+** Not every contributor is a Git expert, do not request changes in the commit history unless it is necessary
+** Core maintainers can squash PRs during the merge.
+   If you feel this is important, use the `squash-me-please` label
+** We want to keep pull requests atomic when possible (one feature / fix - separate commit),
+   but we can live without it if there is no need to backport changes to the stable baseline.
+
+== Merge process
+
+=== Common merge process
+
+**Step 1: Maintainer checklist**. Merge process can be initiated once a pull request matches the requirements:
+
+* Pull request is compliant wuth requirements to submitters (see the link:/.github/PULL_REQUEST_TEMPLATE.md[pull request template])
+* There are at least 2 approvals for the pull request and no outstanding requests for change
+* Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
+* Changelog entries in the PR title and/or `Proposed changelog entries` are correct
+* Proper changelog labels are set so that the changelog can be generated automatically
+* If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set
+  and there is a `Proposed upgrade guidelines` section in the PR title.
+  (link:https://github.com/jenkinsci/jenkins/pull/4387[example])
+* It would make sense to backport the change to LTS, a Jira issue should exist and be labeled as `lts-candidate`
+
+**Step 2: Marking for merge**. Once the checklist is passed,
+a Core PR Reviewer or a Maintainer can mark the pull request for merge.
+
+* `ready-for-merge` label is set
+* An explicit comment is added to the pull request so that other repository watchers are notified.
+  Example: `Thanks to all contributors! We consider this change as read to be merged towards the next weekly release. It may be merged after a 24hours if there is no negative feedback`
+
+**Step 3: Merge!**.
+A Core Maintainer merges the change after a timeout (if needed).
+After that the change will be landed in the next weekly release unless it gets reverted later.
+LTS Backporting, if needed, will be handled separately by the release team.
+
+=== Exceptions
+
+* Jenkins Security Team uses a different process for security issue fixes.
+  They are reviewed and integrated by the Security team in private repositories.
+  Security hardening and enhancements go through the standard process.
+* Release Team members are permitted to bypass the review/merge process if and only if a change is needed to unblock the security release.
+  Common review process is used otherwise.
+* 24 hours timeout after `ready-for-merge` is not required for...
+//TODO(oleg_nenashev): Define "trivial" better to avoid loopholes
+** trivial changes like typo fixes or minor Javadoc improvements
+** changes which do not affect the production code: Jenkinsfile tweaks, tools inside the repo, etc. 
+
```

Comment:
may not need to be stated, but possibly could add:
```suggestion
** broken master build

```

Reply:
Agree. Cleaning up after bad merges should be expedited to unblock all other PRs and development.

Reply:
I'm on board with this given that we were bitten by this exact scenario recently in #4467